examples/joins/README.md

```diff
@@ -1,193 +1,231 @@
+
+See the `transform.build.language` field in our pipeline's (inner_join.json) specifications :
+```shell
+    "transform": {
+        "build": { 
+        "language":"python",
+        "path": "./src/inner"
+        }
+    }
+```
+A more classic way, once 'production-ready,' would be to reference your [built Docker image](https://docs.pachyderm.com/latest/how-tos/developer-workflow/working-with-pipelines/#step-2-build-your-docker-image) as in the example below:
+```shell
+    "transform": {
+        "cmd": [ "python3", "/edges.py" ],
+        "image": "pachyderm/opencv"
+    }
+```
+
+***Step 4*** - Let's have a look at our final product: Check the output repository of your pipeline.
+```shell
+$ pachctl list file inner_join@master
+```
+You should see our 2 expected text files. 
+
+![output_repository](./img/pachctl_list_file_inner_join_master.png)
+
+Now for a visual confirmation of their content:
+```shell
+$ pachctl get file inner_join@master:/02108.txt
+```
+>![pach_logo](./img/pach_logo.svg) Want to take this example to the next level? Practice using joins AND [groups](Add link to group). You can create a 2 steps pipeline that will group Returns and Purchases by storeID then join the output repo with Stores to agregate by location. 
+
+
+## 5. Example 2 - Outer-Join pipeline creation 
+***Goal***
+Well, it depends on what you want to achieve here... We will either get:
+- the zip of **all stores** (listed in our repo or not) **where at least one return was made**
+- all **zip of stores listed** in our repo, **with or without returns**
+- the zip of **all stores** listed or unlisted in our repo, **with or without returns**
+
+The end result will vary depending on where we will focus the outer join. Although the behaviour might slightly differ from SQL because, again, we are working at the file path level, think about a left outer, right outer, and full outer.
+
+1. **Pipeline input repositories**: `stores` and `returns` - Outer join by STOREID: 3 scenarios
+2. **Pipeline**: Executes a python code reading the `zipcode` (if any...) in the matching STOREIDx.txt file and appending the matched return file's content (if any...) to a text file named after the zip code. 
+3. **Pipeline output repository**: `outer_join`- list of text files named after the stores' zip codes. What list will be produced depends on where the outer join's focus was put on. We will get into the detail of each case to help you figure out the best scenario for you.
+
```

Comment:
I feel like there's something missing before you jump into the diagram. Specifically, I think we should be showing the pipeline spec input, and telling the user that "outer is a boolean field that can be set independently on each input repo. Setting outer: true means that files in that repo, even if they don't match, will still be processed as a datum."

Also, I'm not sure we need to cover all 3 out join scenarios at once. One idea might be to have the default be outer: true on both repos (like it is) and have them run just that and show the diagram. Then, have Next Steps be:

"experiment with only having outer set on one repo or the other. This can mirror real world situations such as...<as you describe above>. We've also drawn the diagrams for you to help visualize what is happening in each of the scenarios where only one of the inputs has `outer:true` set."

What do you think?

Reply:
I like the idea of the other outer-join options presented in "Want to take this example to the next level?" with the expected results listed in each case. Will add the pipeline spec for more clarity. I initially thought that it belongs to the documentation itself but there is no harm in adding it here. Some people would rather learn by example before taking a look at the doc.